src/Utilities/distance_to_set.jl

```diff
@@ -193,3 +193,46 @@ function distance_to_set(
     _check_dimension(x, set)
     return zero(T)
 end
+
+function distance_to_set(
+    ::ProjectionUpperBoundDistance,
+    x::AbstractVector{T},
+    set::MOI.SecondOrderCone,
+) where {T<:Real}
+    _check_dimension(x, set)
+    # Projections come from:
+    # Parikh, N., & Boyd, S. (2014). Proximal algorithms. Foundations and
+    # trends in Optimization, page 184, section 6.3.2.
+    t, rhs = x[1], LinearAlgebra.norm(@views x[2:end])
+    if rhs <= -t  # Projection to the point (0, [0])
```

Comment:
You mean (t, [0]) ?

Reply:
No, I think I mean `(0, [0])`? I just followed Section 6.3.2 here: https://web.stanford.edu/~boyd/papers/pdf/prox_algs.pdf